zkevm-circuits/src/evm_circuit/execution/extcodehash.rs

```diff
@@ -0,0 +1,148 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        param::N_BYTES_ACCOUNT_ADDRESS,
+        step::ExecutionState,
+        table::{AccountFieldTag, CallContextFieldTag},
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition::Delta},
+            from_bytes, Cell, RandomLinearCombination, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use eth_types::{evm_types::GasCost, Field, ToAddress, ToLittleEndian, ToScalar, U256};
+use halo2_proofs::{circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct ExtcodehashGadget<F> {
+    same_context: SameContextGadget<F>,
+    external_address: RandomLinearCombination<F, N_BYTES_ACCOUNT_ADDRESS>,
+    tx_id: Cell<F>,
+    previously_accessed: Cell<F>,
+    external_code_hash: Cell<F>,
```

Comment:
This should be `Word<F>`, right? As the code hash can be more than 254 bits

Reply:
The code hash is 256 bits, but the constraints are all expressed in terms of its RLC encoding. The code hash is RLC encoded for the RW table here: https://github.com/scroll-tech/zkevm-circuits/blob/e2a746f632f67e905fde5390e604c491cbc367fc/zkevm-circuits/src/evm_circuit/witness.rs#L647 and `self.external_code_hash` is RLC encoded in the `assign_exec_step` step in this gadget as well.

Reply:
Correct!